src/vellum/workflows/nodes/displayable/bases/prompt_deployment_node.py

```diff
@@ -87,53 +87,73 @@ def _get_prompt_event_stream(self, ml_model_fallback: Optional[str] = None) -> I
-            ):
-                for ml_model_fallback in self.ml_model_fallbacks:
-                    try:
-                        prompt_event_stream = self._get_prompt_event_stream(ml_model_fallback=ml_model_fallback)
-                        next(prompt_event_stream)
-                        break
-                    except ApiError:
-                        continue
+
+        if prompt_event_stream is None:
+            try:
+                prompt_event_stream = self._get_prompt_event_stream()
+                next(prompt_event_stream)
+            except ApiError as e:
+                if (
+                    e.status_code
+                    and e.status_code < 500
+                    and self.ml_model_fallbacks is not OMIT
+                    and self.ml_model_fallbacks is not None
+                ):
+                    prompt_event_stream = self._retry_prompt_stream_with_fallbacks()
                 else:
-                    self._handle_api_error(
-                        ApiError(
-                            body={
-                                "detail": f"Failed to execute prompts with these fallbacks: {self.ml_model_fallbacks}"
-                            },
-                            status_code=400,
-                        )
-                    )
-            else:
-                self._handle_api_error(e)
+                    self._handle_api_error(e)
 
         outputs: Optional[List[PromptOutput]] = None
-        for event in prompt_event_stream:
-            if event.state == "INITIATED":
-                continue
-            elif event.state == "STREAMING":
-                yield BaseOutput(name="results", delta=event.output.value)
-            elif event.state == "FULFILLED":
-                outputs = event.outputs
-                yield BaseOutput(name="results", value=event.outputs)
-            elif event.state == "REJECTED":
-                workflow_error = vellum_error_to_workflow_error(event.error)
-                raise NodeException.of(workflow_error)
+        if prompt_event_stream is not None:
+            for event in prompt_event_stream:
+                if event.state == "INITIATED":
+                    continue
+                elif event.state == "STREAMING":
+                    yield BaseOutput(name="results", delta=event.output.value)
+                elif event.state == "FULFILLED":
+                    outputs = event.outputs
+                    yield BaseOutput(name="results", value=event.outputs)
+                elif event.state == "REJECTED":
+                    if (
+                        event.error
+                        and event.error.code == WorkflowErrorCode.PROVIDER_ERROR.value
+                        and self.ml_model_fallbacks is not OMIT
+                        and self.ml_model_fallbacks is not None
+                    ):
+                        try:
+                            fallback_stream = self._retry_prompt_stream_with_fallbacks()
```

Comment:
[oos] Something tells me that we're going to want to try to consolidate these two retry calls together. I mention two test cases below that we should iterate on so that we validate that they work as expected, that may be down stream of these retry calls not being consolidated